jobs/pacman-cloud-discovery/src/main/java/com/tmobile/cso/pacman/inventory/util/InventoryUtil.java

```diff
@@ -3172,6 +3104,52 @@ public static Map<String, List<DescribeHubResult>> fetchSecurityHub(BasicSession
 		}
 		return securityHubMap;
 	}
+
+	public static Map<String,List<LaunchTemplateVH>> fetchLaunchTemplates(BasicSessionCredentials temporaryCredentials, String skipRegions,String accountId,String accountName){
+		Map<String,List<LaunchTemplateVH>> launchTemplates = new LinkedHashMap<>();
+		String expPrefix = InventoryConstants.ERROR_PREFIX_CODE+accountId + "\",\"Message\": \"Exception in fetching info for resource in specific region\" ,\"type\": \"Launch Template\" , \"region\":\"" ;
+		AmazonEC2 ec2Client ;
+		List<LaunchTemplateVH> launchTemplateVHList = new ArrayList<>();
+		for(Region region : RegionUtils.getRegions()) {
+			try{
+				if(!skipRegions.contains(region.getName())){
+					ec2Client = AmazonEC2ClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(temporaryCredentials)).withRegion(region.getName()).build();
+					DescribeLaunchTemplatesRequest request = new DescribeLaunchTemplatesRequest();
+					DescribeLaunchTemplatesResult descLaunchTempResult ;
+					descLaunchTempResult=ec2Client.describeLaunchTemplates(request);
+					final AmazonEC2 finalEc2Client = ec2Client;
+					descLaunchTempResult.getLaunchTemplates().forEach(template -> {
+						try {
+							DescribeLaunchTemplateVersionsRequest versionsRequest = new DescribeLaunchTemplateVersionsRequest()
+									.withLaunchTemplateId(template.getLaunchTemplateId());
+
+							DescribeLaunchTemplateVersionsResult describeVersionsResult = finalEc2Client.describeLaunchTemplateVersions(versionsRequest);
+
+							describeVersionsResult.getLaunchTemplateVersions().forEach(version -> {
+								LaunchTemplateVH launchTemplateVH = new LaunchTemplateVH();
+								launchTemplateVH.setLaunchTemplateId(template.getLaunchTemplateId());
+								launchTemplateVH.setLaunchTemplateName(template.getLaunchTemplateName());
+								launchTemplateVH.setImageId(version.getLaunchTemplateData().getImageId());
+								launchTemplateVH.setSecurityGroupIds(version.getLaunchTemplateData().getSecurityGroupIds());
+								launchTemplateVHList.add(launchTemplateVH);
+							});
+						} catch (Exception e) {
+							log.warn("Error fetching versions for Launch Template: " + template.getLaunchTemplateName());
+						}
+					});
+
+					if (!launchTemplateVHList.isEmpty()) {
+						log.debug(InventoryConstants.ACCOUNT + accountId + " Type: Launch Template " + region.getName() + " >> " + launchTemplateVHList.size());
+						launchTemplates.put(accountId + delimiter + accountName + delimiter + region.getName(), launchTemplateVHList);
+					}
+				}
+			}catch(Exception e){
+				log.warn(expPrefix+ region.getName()+InventoryConstants.ERROR_CAUSE +e.getMessage()+"\"}");
```

Comment:
it should be log.error

Reply:
For every collector in aws we are use log.warn only